src/main/java/se/liu/ida/hefquin/engine/data/impl/VocabularyMappingImpl.java

```diff
@@ -0,0 +1,116 @@
+package se.liu.ida.hefquin.engine.data.impl;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Set;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.query.Query;
+import org.apache.jena.query.QueryExecution;
+import org.apache.jena.query.QueryExecutionFactory;
+import org.apache.jena.query.QueryFactory;
+import org.apache.jena.query.QuerySolution;
+import org.apache.jena.query.ResultSet;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.reasoner.TriplePattern;
+import org.apache.jena.riot.RDFDataMgr;
+
+import se.liu.ida.hefquin.engine.data.VocabularyMapping;
+
+public class VocabularyMappingImpl implements VocabularyMapping{
+	
+	protected final Model vocabularyMapping;
+	
+	public VocabularyMappingImpl() {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+	}
+	
+	
+	// Source: https://jena.apache.org/documentation/io/rdf-input.html
+	public VocabularyMappingImpl(final String rdfFile) {
+		vocabularyMapping = RDFDataMgr.loadModel(rdfFile); //.nt file for N-Triple
+	}
+	
+	public VocabularyMappingImpl(final Set<TriplePattern> tps) {
+		vocabularyMapping = ModelFactory.createDefaultModel();
+		Iterator<TriplePattern> i = tps.iterator();
+		while(i.hasNext()) {
+			TriplePattern tp = i.next();
+			Resource s = vocabularyMapping.createResource(tp.getSubject().toString());
+			Property p = vocabularyMapping.createProperty(tp.getPredicate().toString());
+			RDFNode o = vocabularyMapping.createResource(tp.getObject().toString());
+			Statement statement = vocabularyMapping.createStatement(s, p, o);
+			vocabularyMapping.add(statement);
+		}
+	}
+
+	@Override
+	//TODO: Which data structure to use? Unions and intersections also need to be represented
+	public Set<TriplePattern> translateTriple(TriplePattern tp) {
+		Set<TriplePattern> results = translateSubject(tp);
+		return results;
+	}
+	
+	private Set<TriplePattern> translateSubject(TriplePattern tp){
+		Set<TriplePattern> results = Collections.emptySet();
+		if (tp.getSubject().isVariable()) {
+			results.add(tp);
+			return results;
+		}
+		Node p = NodeFactory.createVariable("p");
+		Node o = NodeFactory.createVariable("o");
+		TriplePattern tpQuery = new TriplePattern(tp.getSubject(), p, o);
+		Set<TriplePattern> mappings = getMappings(tpQuery);
+		
+		/* OR ?
+		 * Selector selector = new SimpleSelector(tp.getSubject(), null, null);
+		 * Iterator i = vocabularyMapping.listStatements(selector);
+		 */
+		
+		Iterator<TriplePattern> i = mappings.iterator();
+		while (i.hasNext()) {
+			TriplePattern m = i.next();
+			if (m.getPredicate().toString() == "owl:sameAs") {
```

Comment:
This won't work. Notice that 'owl:sameAs' is just a short form for the full URI 'http://www.w3.org/2002/07/owl#sameAs' and can be used only in contexts in which the prefix 'owl' is declared, which is not the case here. So, you need to use the full URI for the comparison.

Having said that, the Jena API has helper classes with static member variable for the most common vocabularies, including the OWL vocabulary and the RDFS vocabulary. You find them in the Java package `org.apache.jena.vocabulary` of Jena. For instance, the one of 'owl:sameAs' is `OWL.sameAs`. Using these ones helps to avoid making errors with wrongly types full URIs.

Reply:
Just to chime in, when comparing Strings, you need to use the equals function instead of the `==` operator. Something ike this:
```
m.getPredicate().toString().equals(OWL.sameAs.toString())
```

Reply:
Good point Sebastian!

In fact, when comparing Java objects for equivalence, it is always a good idea to use 'equals'.  == is useful mostly for comparing atomic values.